cfstore/interface.py

```diff
@@ -440,22 +440,57 @@ def delete_file_from_collection(self, collection, file):
         self.session.commit()
 
         
-    def retrieve_variable(self, **kw):
+    def retrieve_variable(self, key, value):
         """ Retrieve variable by arbitrary property"""
         queries = []
-        for k,v in kw.items():
-            if k in ['long_name','standard_name','cfdm_size','cfdm_domain','cell_methods']:
-                queries.append(getattr(Variable,k) == v)
-            else:
-                queries.append(Variable.with_other_attributes(k,v))
+        if key in ['long_name','standard_name','cfdm_size','cfdm_domain','cell_methods']:
+            queries.append(getattr(Variable,key) == value)
+        else:
+            queries.append(Variable.with_other_attributes(key,value))
+        if key == 'in_files':
+            queries.append([value == k for k in Variable.in_files])
         if len(queries) == 0:
             raise ValueError('No query received for retrieve variable')
         elif len(queries) == 1:
             results = self.session.query(Variable).filter(queries[0]).all()
         else:
             results = self.session.query(Variable).filter(and_(*queries)).all()
+        if key == "all":
+            results = self.session.query(Variable).all()
         return results
 
+    def retrieve_variable_query(self, key, value, query):
+        """ Retrieve variable by arbitrary property"""
+        queries = query
+        print(queries)
+        if key in ['long_name','standard_name','cfdm_size','cfdm_domain','cell_methods']:
+            queries.append(getattr(Variable,key) == value)
+        else:
+            queries.append(Variable.with_other_attributes(key,value))
+        if key == 'in_files':
+            queries.append([value == k for k in Variable.in_files])
+        if len(queries) == 0:
+            raise ValueError('No query received for retrieve variable')
+        elif len(queries) == 1:
+            results = self.session.query(Variable).filter(queries[0]).all()
+        else:
+            results = self.session.query(Variable).filter(and_(*queries)).all()
+        return results, queries
+
+    def show_collections_with_variable(self,variable):
```

Comment:
It would be helpful to be clear whether the argument variable is expected to be a string (name of variable), or a variable instance. I think it's the latter here, but it's an important distinction.